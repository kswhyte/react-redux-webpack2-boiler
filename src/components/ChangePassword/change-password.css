.change-password-wrapper {
    & .change-password-spacer {
        margin-top: 120px;
    }
}