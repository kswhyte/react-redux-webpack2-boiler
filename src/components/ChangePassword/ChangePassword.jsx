import React from 'react';
import { PropTypes } from 'prop-types';

import './change-password.css';

const propTypes = {
  changePassword: PropTypes.func
};

const checkPassword = () => {
  const loginChangePassword = escape(document.querySelector('#loginChangePassword').value);
  const loginChangePasswordConfirm = escape(document.querySelector('#loginChangePasswordConfirm').value);
  if (loginChangePassword !== loginChangePasswordConfirm) {
    console.log("Ain't no gooood");
  }
  if (loginChangePassword === loginChangePasswordConfirm) {
    console.log("Looks good");
  }
}

const submitPasswordChange = (submitCallback) => {
  const loginChangePassword = escape(document.querySelector('#loginChangePassword').value);
  const loginChangePasswordConfirm = escape(document.querySelector('#loginChangePasswordConfirm').value);
  if (loginChangePassword !== loginChangePasswordConfirm) {
    console.log("Ain't no gooood 2");
  }
  if (loginChangePassword === loginChangePasswordConfirm) {
    console.log("Looks good 2");
    submitCallback(loginChangePassword);
  }
}

const ChangePassword = () => {
  return (
    <div className="change-password-wrapper card">
        <img src="images/pattern-dots.png" className="dots-pattern" />
        <div className="row">
          <h2>UPDATE PASSWORD</h2>
          <div className="page-instructions col-xs-offset-4 col-xs-4">
            <p>Please enter and confirm your new password below.</p>
          </div>
        </div>
        <form
          onSubmit={e => {
            e.preventDefault();
            submitPasswordChange();
          }}
        >
          <div className="row change-password-spacer">
            <div className="col-sm-offset-1 col-sm-5 input-row">
              <label forHtml="loginChangePassword">NEW PASSWORD</label>
              <input
                id="loginChangePassword"
                type="password"
                placeholder="Your New Password"
                required
              />
            </div>
            <div className="col-sm-5 input-row">
              <label forHtml="loginChangePasswordConfirm">CONFIRM PASSWORD</label>
              <input
                id="loginChangePasswordConfirm"
                type="password"
                placeholder="Confirm Password"
                label="Confirm Password"
                onBlur={checkPassword}
                required
              />
            </div>
          </div>
          <div className="row change-password-spacer">
            <div className="col-xs-offset-3 col-xs-6">
              <button className="primary-btn">
                CHANGE PASSWORD
              </button>
            </div>
          </div>
        </form>
      </div>
  )
}

ChangePassword.propTypes = propTypes;

export default ChangePassword;