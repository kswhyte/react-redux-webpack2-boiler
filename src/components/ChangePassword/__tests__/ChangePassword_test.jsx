import React from 'react';
// import { storiesOf, action, linkTo } from '@kadira/storybook';
import ChangePassword from '../ChangePassword';

// import { specs, describe, it } from 'storybook-addon-specifications';

import { storiesOf, describe, it, specs } from '../../../../.storybook/facade';
import { mount } from 'enzyme';
import expect from 'expect';

const stories = storiesOf('ChangePassword', module);

stories.add('ChangePassword Page Story', () => {
  const changePasswordStory = (
      <ChangePassword />
  );

  specs(() =>
    describe('Show a successful alert', () => {
      it('Should render the ChangePassword component without crashing', () => {
        let output = mount(changePasswordStory);
        expect(output.find('.change-password-wrapper').length).toEqual(1);
      });
      it('Should have two input fields', () => {
        let output = mount(changePasswordStory);
        expect(output.find('input').length).toEqual(2);
      });
      it('Should have a header element that says "Update Password"', () => {
        let output = mount(changePasswordStory);
        expect(output.find('h2').text()).toContain('UPDATE PASSWORD');
      });
      it('Should have a button to submit"', () => {
        let output = mount(changePasswordStory);
        expect(output.find('button').text()).toContain('CHANGE PASSWORD');
      });
    })
  );
  return changePasswordStory;
});