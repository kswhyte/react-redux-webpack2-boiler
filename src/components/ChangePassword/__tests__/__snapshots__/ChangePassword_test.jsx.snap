// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangePassword Page Story 1`] = `
<div
  className="change-password-wrapper card"
>
  <img
    className="dots-pattern"
    src="images/pattern-dots.png"
  />
  <div
    className="row"
  >
    <h2>
      UPDATE PASSWORD
    </h2>
    <div
      className="page-instructions col-xs-offset-4 col-xs-4"
    >
      <p>
        Please enter and confirm your new password below.
      </p>
    </div>
  </div>
  <div
    className="alert-row"
  />
  <form
    onSubmit={[Function]}
  >
    <div
      className="row change-password-spacer"
    >
      <div
        className="col-sm-offset-1 col-sm-5 input-row"
      >
        <label
          htmlFor="loginChangePassword"
        >
          NEW PASSWORD
        </label>
        <input
          id="loginChangePassword"
          placeholder="Your New Password"
          required={true}
          type="password"
        />
      </div>
      <div
        className="col-sm-5 input-row"
      >
        <label
          htmlFor="loginChangePasswordConfirm"
        >
          CONFIRM PASSWORD
        </label>
        <input
          id="loginChangePasswordConfirm"
          label="Confirm Password"
          onBlur={[Function]}
          placeholder="Confirm Password"
          required={true}
          type="password"
        />
      </div>
    </div>
    <div
      className="row change-password-spacer"
    >
      <div
        className="col-xs-offset-3 col-xs-6"
      >
        <button
          className="primary-btn"
        >
          CHANGE PASSWORD
        </button>
      </div>
    </div>
  </form>
</div>
`;
